src/entities/plugin.ts

```diff
@@ -0,0 +1,27 @@
+// The plugin entity
+export interface Plugin {
+  plugin_id: number,
+  plugin_version: string,
+  plugin_name: string
+}
+
+export class PluginImpl implements Plugin {
+  public plugin_id: number;
+  public plugin_version: string;
+  public plugin_name: string;
+
+  constructor(params: Plugin) {
+    this.plugin_id = params.plugin_id;
+    this.plugin_version = params.plugin_version;
+    this.plugin_name = params.plugin_name;
+  }
+}
+
+export async function buildPluginPayload(plugin: Plugin) {
```

Comment:
Could we make this an "instance method" (sorry, Ruby nomenclature).

Like 
```
buildPayload() {
  return {
     schema: "iglu:com.software/plugin/jsonschema/1-0-0",
     data: {
       ...this
     }
}
```

So when its being used we can do:
```
const plugin: Plugin = new Plugin(params);   // if we change the interface and class names to not have the Impl
const _pluginPayload = await plugin.buildPayload();
```

Reply:
yes, totally agree, will update